mlir/lib/Conversion/TosaToLinalg/TosaToLinalgPipeline.cpp

```diff
@@ -0,0 +1,66 @@
+//===- TosaToLinalgPipeline.cpp - Lowering Tosa to Linalg Dialect ---------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This transformation pass legalizes Tosa operations to the Linalg dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/TosaToLinalg/TosaToLinalg.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/Dialect/Tosa/Transforms/Passes.h"
+#include "mlir/Dialect/Tosa/Utils/QuantUtils.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "mlir/Transforms/Passes.h"
+
+namespace mlir {
+#define GEN_PASS_DEF_TOSATOLINALGPIPELINE
+#include "mlir/Conversion/Passes.h.inc"
+} // namespace mlir
+
+using namespace mlir;
+
+namespace {
+struct TosaToLinalgPipeline
+    : public impl::TosaToLinalgPipelineBase<TosaToLinalgPipeline> {
+public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry
+        .insert<arith::ArithDialect, linalg::LinalgDialect, math::MathDialect,
+                tensor::TensorDialect, scf::SCFDialect>();
+  }
+
+  void runOnOperation() override {
+    OpPassManager pm("builtin.module");
+
+    TosaToLinalgOptions tosaToLinalgOptions;
+
+    tosa::addTosaToLinalgPasses(pm, tosaToLinalgOptions,
+                                /* validationOptions = */
+                                {tosa::TosaProfileEnum::BaseInference,
+                                 /* StrictOperationSpecAlignment = */ true,
+                                 tosa::TosaLevelEnum::EightK});
+
+    if (failed(runPipeline(pm, getOperation())))
+      signalPassFailure();
+  }
```

Comment:
Why isn't this just a registered pipeline instead of a pass? https://mlir.llvm.org/docs/PassManagement/#pass-pipeline-registration

Reply:
revised as a registered pipeline now